wagtail/admin/templates/wagtailadmin/base.html

```diff
@@ -5,7 +5,8 @@
     {% slim_sidebar_enabled as slim_sidebar_enabled %}
     <template data-wagtail-sidebar-branding-logo>{% block branding_logo %}{% endblock %}</template>
     {% if slim_sidebar_enabled %}
-    <aside id="wagtail-sidebar" data-wagtail-sidebar data-props="{% menu_props %}"></aside>
+    <script id="wagtail-sidebar-props" type="application/json">{% autoescape off %}{% menu_props %}{% endautoescape %}</script>
```

Comment:
Is HTML injection possible here? Say, in a menu item name someone was able to write: ``</script><script>doSomethingBad(); //``, would that initial ``</script>`` appear as-is causing this script tag to be closed?

Reply:
Good catch! I was trying to replicate Django’s `json_script`, turns out it also does HTML escaping. So I’ve switched `menu_props` to use it directly, and this will prevent this type of XSS.